Engine_Revit_UI/Modify/SetParameter.cs

```diff
@@ -133,9 +133,49 @@ public static bool SetParameter(this Element element, BuiltInParameter builtInPa
 
         public static bool SetParameter(this Parameter parameter, object value, Document document = null)
         {
+            // Skip null and read-only parameters
             if (parameter == null || parameter.IsReadOnly)
                 return false;
 
+            // Workset parameters
+            // Filter for worksets
+            if (parameter.Id.IntegerValue == (int)BuiltInParameter.ELEM_PARTITION_PARAM)
+            {
+                // Skip empty values
+                if (string.IsNullOrWhiteSpace(value as string))
+                    return false;
+
+                // Find all user worksets
+                FilteredWorksetCollector worksets =
+                    new FilteredWorksetCollector(document)
+                    .OfKind(WorksetKind.UserWorkset);
+
+                // Does a workspace with the specified name exist?
+                foreach (Workset workset in worksets)
+                {
+                    if (workset.Name == value as string)
```

Comment:
What happens if `value` is not a `string`?

Reply:
Good point, a check should happen at the beginning

Reply:
Relocated to the top